src/content/app/tools/blast/components/blast-input-sequences/BlastInputSequence.tsx

```diff
@@ -128,19 +154,33 @@ const BlastInputSequence = (props: Props) => {
         <Textarea
           ref={textareaRef}
           value={input}
-          className={styles.textarea}
-          placeholder="Paste a nucleotide or protein sequence"
+          className={textareaClasses}
+          placeholder="Paste plain text or FASTA sequence of at least 5 characters"
           onChange={onChange}
           onPaste={onPaste}
+          onFocus={onFocus}
           onBlur={onBlur}
           resizable={false}
         />
-        {!input && <Upload transformTo="text" onUpload={onFileDrop} />}
+        {!input && !isFocused && (
+          <Upload transformTo="text" onUpload={onFileDrop} />
+        )}
       </div>
     </div>
   );
 };
 
+const checkSequenceValidity = (
```

Comment:
Can this function be moved to `sequenceValidators.ts`?

Reply:
I've added a function that combines the checks `hasSufficientLength` and  `hasValidSequenceCharacters` to `sequenceValidators.ts`; but the `checkSequenceValidity` function above contains the cleanup logic that belongs in the input box.